### PR TITLE
Enable mgpu unit tests for rocm

### PR DESCRIPTION
**Pull request by jithunnair-amd**

### CODE REVIEW

**Comment by jithunnair-amd** [comment]
@bddppq @pietern Can someone from your team please review this PR? It's meant to enable the test_c10d and test_nccl unit test suites for ROCm, while skipping the ones that don't work currently. I haven't enabled test_distributed unit test suite for ROCm yet, because it will need skipping of gloo and mpi backend for ROCm (as they're not fully tested yet), and will also need changes to `test_distributed.py` to properly skip the tests without error. I wasn't sure if I should do that since it seems like there might be some refactoring happening to eventually use similar code that exists in `common_distributed.py`

**Review by bddppq — APPROVED** [review]
LG

File: test/common_distributed.py [hunk 1]
```diff
@@ -91,6 +92,16 @@ def requires_mpi():
     )
 
 
+def skip_for_rocm(func):
```
Comment (bddppq) [inline test/common_distributed.py]: nit: I feel it's better to name it "skip_if_rocm" to be more consistent with the `skipIfRocm` `decorator`

Comment (jithunnair-amd) [reply test/common_distributed.py]: I agree. Will update.

**Pull request closed without merging**